src/app/clusters/scenes-server/scenes-server.cpp

```diff
@@ -0,0 +1,993 @@
+
+    return CHIP_NO_ERROR;
+}
+
+// CommandHanlerInterface
+void ScenesServer::InvokeCommand(HandlerContext & ctxt)
+{
+    mPath = ctxt.mRequestPath;
+
+    switch (mPath.mCommandId)
+    {
+    case Commands::AddScene::Id:
+        HandleCommand<Commands::AddScene::DecodableType>(
+            ctxt, [this](HandlerContext & ctx, const auto & req) { HandleAddScene(ctx, req); });
+        return;
+    case Commands::ViewScene::Id:
+        HandleCommand<Commands::ViewScene::DecodableType>(
+            ctxt, [this](HandlerContext & ctx, const auto & req) { HandleViewScene(ctx, req); });
+        return;
+    case Commands::RemoveScene::Id:
+        HandleCommand<Commands::RemoveScene::DecodableType>(
+            ctxt, [this](HandlerContext & ctx, const auto & req) { HandleRemoveScene(ctx, req); });
+        return;
+    case Commands::RemoveAllScenes::Id:
+        HandleCommand<Commands::RemoveAllScenes::DecodableType>(
+            ctxt, [this](HandlerContext & ctx, const auto & req) { HandleRemoveAllScenes(ctx, req); });
+        return;
+    case Commands::StoreScene::Id:
+        HandleCommand<Commands::StoreScene::DecodableType>(
+            ctxt, [this](HandlerContext & ctx, const auto & req) { HandleStoreScene(ctx, req); });
+        return;
+    case Commands::RecallScene::Id:
+        HandleCommand<Commands::RecallScene::DecodableType>(
+            ctxt, [this](HandlerContext & ctx, const auto & req) { HandleRecallScene(ctx, req); });
+        return;
+    case Commands::GetSceneMembership::Id:
+        HandleCommand<Commands::GetSceneMembership::DecodableType>(
+            ctxt, [this](HandlerContext & ctx, const auto & req) { HandleGetSceneMembership(ctx, req); });
+        return;
+    case Commands::EnhancedAddScene::Id:
+        HandleCommand<Commands::EnhancedAddScene::DecodableType>(
+            ctxt, [this](HandlerContext & ctx, const auto & req) { HandleEnhancedAddScene(ctx, req); });
+        return;
+    case Commands::EnhancedViewScene::Id:
+        HandleCommand<Commands::EnhancedViewScene::DecodableType>(
+            ctxt, [this](HandlerContext & ctx, const auto & req) { HandleEnhancedViewScene(ctx, req); });
+        return;
+    case Commands::CopyScene::Id:
+        HandleCommand<Commands::CopyScene::DecodableType>(
+            ctxt, [this](HandlerContext & ctx, const auto & req) { HandleCopyScene(ctx, req); });
+        return;
+    }
+}
+
+// AttributeAccessInterface
+CHIP_ERROR ScenesServer::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    uint8_t value             = 0;
+    uint16_t endpoinTableSize = 0;
+    ReturnErrorOnFailure(
+        StatusIB(ToInteractionModelStatus(Attributes::SceneTableSize::Get(aPath.mEndpointId, &endpoinTableSize))).ToChipError());
+
+    // Get Scene Table Instance
+    scenes::SceneTable<scenes::ExtensionFieldSetsImpl> * sceneTable;
+
+    switch (aPath.mAttributeId)
+    {
+    case Attributes::SceneCount::Id:
```

Comment:
Please add SceneCount and RemainingCapacity to the lists of attribute-access-interface-only attributes in `src/app/zap-templates/zcl/zcl.json` and `src/app/zap-templates/zcl/zcl-with-test-extensions.json`.  Followup is fine for this.

Reply:
Added in .json